redux-offline/redux-offline
# Adds Initial: Jest setup, testing utilities, tests

* Add /src/__tests__ directory

    * Add Redux-Mock-Store dev dependency

    * Add jest-setup.js file at project root

    * Add tests for middleware, createOfflineStore, config, and batch

@jevakallio , Alright, I've got these updated to test the newest version of the code.  I have one test that I'm currently skipping because something isn't mocking correctly which is proving to be a little bit tricky to debug, so I figured I would just submit the PR now because I'm not sure how long it will take to figure it out.

As I said in my last PR, I just have a few tests in so far which a fairly straight forward unit tests, but if we get these to a point where you're happy with the direction and want to move forward. I'll finish off the unit tests and try to get in some full integration tests as well.

One thing you might notice is that some of the formatting is updated on a few of the non-test files.  That was a mistake initially because my editor auto-formats when I switch panes which I have set up to use Prettier.  I was initially going to pull out these changes, but I figured if using Prettier is the standard you want to use for formatting, it might make sense to leave the changes in for consistency.  That being said, I don't want to create any merge conflict headaches for anyone, so if need be, I can definitely revert them to the original formatting.
